src/checkbox.jsx

```diff
@@ -10,15 +10,61 @@ import ThemeManager from './styles/theme-manager';
 const Checkbox = React.createClass({
 
   propTypes: {
+    /**
+     * Checkbox is checked if true.
+     */
     checked: React.PropTypes.bool,
+
+    /**
+     * The SvgIcon to use for the checked state.
+     * This is useful to create icon toggles.
+     */
     checkedIcon: React.PropTypes.element,
+
+    /**
+     * The default state of our checkbox component.
+     */
     defaultChecked: React.PropTypes.bool,
+
+    /**
+     * Disabled if true.
+     */
     disabled: React.PropTypes.bool,
+
+    /**
+     * Overrides the inline-styles of the icon element.
+     */
     iconStyle: React.PropTypes.object,
+
+    /**
+     * Where the label will be placed next to the checkbox.
+     */
     labelPosition: React.PropTypes.oneOf(['left', 'right']),
+
+    /**
+     * Overrides the inline-styles of the Checkbox element label.
+     */
     labelStyle: React.PropTypes.object,
+
+    /**
+     * Callback function that is fired when the checkbox is checked.
+     */
     onCheck: React.PropTypes.func,
+
+    /**
+     * Override the inline-styles of the root element.
+     */
+    style: React.PropTypes.object,
+
+    /**
+     * The SvgIcon to use for the unchecked state.
+     * This is useful to create icon toggles.
+     */
     unCheckedIcon: React.PropTypes.element,
+
+    /**
+     * ValueLink for when using controlled checkbox.
+     */
     valueLink: React.PropTypes.object,
```

Comment:
I think we should do it sooner. to avoid a ton of deprecation warnings when they do. besides we should at the very least be ready for it. We should provide value and onChange for components with only valueLink